### PR TITLE
Refer-to to not to modify uri if it contains host address

### PR DESCRIPTION
**Pull request by vinoddh-kore**

### CODE REVIEW

**Comment by davehorton** [comment]
@vinoddh-kore  I can't accept this PR as is because its not quite that simple. 

In fact, this was a feature added to support transferring calls to the PSTN on twilio.  Let's say I create an elastic sip trunk on twilio and in the termination panel on twilio I set the dns name of the termination to be "my-trunk.pstn.twilio.com".  I then enable support for REFER **and** support for transferring to PSTN numbers

If I then receive a call from Twilio, answer it and later try to transfer by sending this REFER:

```
REFER sip:+919972494697@172.18.212.166:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 35.158.245.46;rport;branch=z9hG4bKrr9NQaS33vm7e
Route: <sip:54.171.127.194;lr>
Max-Forwards: 70
From: <sip:+441782986064@35.158.245.46>;tag=KH69vy90tpH1S
To: <sip:+919972494697@mydomain.pstn.twilio.com>;tag=05360357_c3356d0b_877fcc5c-2ef9-497b-b3f6-d9a7de8018e3
Call-ID: 9c0a86bf49b0b8bf1ddee74377905050@0.0.0.0
CSeq: 2 REFER
Refer-To: <sip:+491745915912@54.171.127.194:5060>
Referred-By: <sip:+919972494697@35.158.245.46>
Content-Length: 0
Contact: <sip:35.158.245.46:5060;transport=udp>  
```

I will get a 202 Accepted to the REFER but I will then get a NOTIFY with a 403 final status on the subsequent call attempt.

On the other hand, if I set up the Refer-To like below (which is what the current implementation will do) the transfer is successful:

```
REFER sip:+919972494697@172.18.197.135:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 35.158.245.46;rport;branch=z9hG4bKye1a2cyrjjtQm
Route: <sip:54.171.127.192;lr>
Max-Forwards: 70
From: <sip:+441782986064@35.158.245.46>;tag=Ba5v77XNNc81F
To: <sip:+919972494697@mydomain.pstn.twilio.com>;tag=52879247_c3356d0b_ef63d06b-ff1a-481d-8e6b-eecc7cbe81fe
Call-ID: b325169a2ebd0355bd95fd11b5040938@0.0.0.0
CSeq: 2 REFER
Refer-To: <sip:+491745915912@my-trunk.pstn.twilio.com:5060>
Referred-By: <sip:+919972494697@35.158.245.46>
Content-Length: 0
Contact: <sip:35.158.245.46:5060;transport=udp>
```

I get a 202 to the REFER and then a NOTIFY with a 200 indicating the transferred call was answered.

This seems to be a twilio thing and I know that several customers rely on it so this PR would be a breaking change.

However, I agree that a change is needed for the cases other than twilio.

I think one possible way to do it would be to change the function that determines which outbound gateway IP to substitute.  Currently it looks like this

```js
  const getOutboundGatewayForRefer = async(voip_carrier_sid) => {
    try {
      const [r] = await pp.query(sqlSelectOutboundGatewayForCarrier, [voip_carrier_sid]);
      if (0 === r.length) return null;

      /* if multiple, prefer a DNS name */
      const hasDns = r.find((row) => row.ipv4.match(/^[A-Za-z]/));
      return hasDns || r[0];
    } catch (err) {
      logger.error({err}, 'getOutboundGatewayForRefer');
    }
  };
```

Here we are looking at the carriers outbound gateways and selecting one, with preference to one that has a DNS name (e.g. in my example above, I want to pick "mytrunk.twilio.com".

I think we could return that to return **only** the outbound gateway that is a DNS, so that if you did not have any outbound gateways defined by a dns name for the carrier, we would leave the uri.host as is.

In other words, change this
```js
   return hasDns || r[0];
```
to this
```js
   return hasDns;
```

Could you back out your change above and test with this suggested change to see if it works for you?

**Comment by davehorton** [comment]
will be fixed in bugfix/issue-63
